app/controllers/webauthn_credentials_controller.rb

```diff
@@ -0,0 +1,55 @@
+class WebauthnCredentialsController < ApplicationController
+  before_action :redirect_to_signin, unless: :signed_in?
+
+  def create
+    @create_options = current_user.webauthn_options_for_create
+
+    session[:webauthn_registration] = { "challenge" => @create_options.challenge }
+
+    render json: @create_options
+  end
+
+  def callback
+    webauthn_credential = build_webauthn_credential
+
+    if webauthn_credential.save
+      redirect_to edit_settings_path
+    else
+      render(
+        json: {
+          message: webauthn_credential.errors.full_messages.to_sentence
+        },
+        status: :unprocessable_entity
+      )
+    end
+  rescue WebAuthn::Error => e
+    render json: { message: e.message }, status: :unprocessable_entity
```

Comment:
Nit: the change in style between L18-22 and this single line layout is a little surprising.

Reply:
I think that was done because of line length in the previous example. I do like the idea of making the layout the same so that one can see that they're doing essentially the same thing, with just a different message, on a quick glance. 

I'm going to extract message into a local variable above so that they can both be single line renders. I think that's a bit neater than them both being multi-line